src/draw.h

```diff
@@ -40,6 +36,27 @@ enum corner_pos {
         _C_LAST = 1 << 5,
 };
 
+struct color {
+        double r;
+        double g;
+        double b;
+        double a;
+};
+
+#define COLOR_UNINIT { -1, -1, -1, -1 }
+#define COLOR_VALID(c) ((c).r >= 0 && (c).g >= 0 && (c).b >= 0 && (c).a >= 0)
```

Comment:
Should we check `c.x > 1` here as well (instead of just when trying to print it)? Cairo colors have to be in the [0, 1] range and thus `COLOR_VALID` should enforce this everywhere.

Reply:
My idea was that either a color was valid or it was initialized with all -1. But yes, maybe an extra check for correctness is fine